packages/pigeon/lib/src/pigeon_lib.dart

```diff
@@ -125,21 +125,33 @@ class FlutterApi {
   const FlutterApi();
 }
 
-/// Metadata to annotate a Pigeon API that wraps a native class.
+/// Metadata to annotate a ProxyAPI.
+///
+/// A ProxyAPI is a generated API for interacting with a native type from Dart.
+/// This includes the generated Dart proxy class and the native type API.
 ///
 /// The abstract class with this annotation groups a collection of Dart↔host
-/// constructors, fields, methods and host↔Dart methods used to wrap a native
-/// class.
+/// constructors, fields, methods and host↔Dart methods used to interact with a
+/// native type.
 ///
-/// The generated Dart class acts as a proxy to a native type and maintains
-/// instances automatically with an `InstanceManager`. The generated host
-/// language class implements methods to interact with class instances or static
-/// methods.
+/// This generates:
+/// 1. A Dart proxy class that handles communication with native type API.
+/// Instances of this proxy class represent instances of the native type.
+/// 2. A native type API which handles communication with the Dart proxy class
+/// and the native type. (e.g. When an instance method of a Dart proxy class is
+/// called, the implementation of the native type API handles calling that
+/// method on the native type.)
+/// 3. An InstanceManager that is a global collection that handles serializable
+/// references to the Dart proxy classes and the native type instances. This
+/// also provides automatic garbage collection of native type instances when its
+/// associated Dart proxy class instance is garbage collected.
 class ProxyApi {
```

Comment:
@tarrinneal @stuartmorgan-g Let me know if this explanation of "what is a ProxyAPI" makes sense.